source/kernel/src/bbq.rs

```diff
@@ -57,6 +72,8 @@ pub async fn new_bidi_channel(
         BBQBidiHandle {
             producer: a_prod,
             consumer: b_cons,
+            our_waitcells: NonNull::new_unchecked(&storage.a_wait as *const BBQWaitCells as *mut BBQWaitCells),
+            other_waitcells: NonNull::new_unchecked(&storage.b_wait as *const BBQWaitCells as *mut BBQWaitCells),
```

Comment:
can't this just be the (safe) `From<&'_ T> for NonNull` impl?
```suggestion
            our_waitcells: NonNull::from(&storage.a_wait),
            other_waitcells: NonNull::from(&storage.b_wait),
```

Reply:
So uh, TIL about `From<&'_ T> for NonNull`.